src/transports/winhttp.c

```diff
@@ -1072,7 +1083,7 @@ static int winhttp_stream_read(
 			/* TODO: extract the username from the url, no payload? */
 			if (t->owner->proxy.credentials) {
 				int cred_error = 1;
-				cred_error = t->owner->proxy.credentials(&t->proxy_cred, t->owner->proxy.url, NULL, allowed_types, NULL);
+				cred_error = t->owner->proxy.credentials(&t->proxy_cred, t->owner->proxy.url, NULL, allowed_types, t->owner->proxy.payload);
```

Comment:
I don't quite understand where this change is coming from yet.  I don't immediately see anything that's using a payload with regards to the digest changes...  Is this a related change, or should it be split out into its own commit?